src/files/examples/pyfingerprint_examples.py

```diff
@@ -0,0 +1,247 @@
+
+def enroll():
+	"""Asks for fingerprint and adds it to database"""
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 1
+		logging.debug("Image read, converting it")
+		f.convertImage(0x01)
+
+		## Checks if finger is already enrolled
+		result = f.searchTemplate()
+		position_number = result[0]
+
+		if ( position_number >= 0 ):
+			print('Template already exists at position #' + str(position_number))
+			exit(0)
+
+		print('Remove finger...')
+		time.sleep(2)
+
+		print('Waiting for same finger again...')
+
+		## Wait that finger is read again
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 2
+		f.convertImage(0x02)
+
+		## Compares the charbuffers
+		if ( f.compareCharacteristics() == 0 ):
+			raise Exception('Fingers do not match')
+
+		## Creates a template
+		f.createTemplate()
+
+		## Saves template at new position number
+		position_number = f.storeTemplate()
+		print('Finger enrolled successfully!')
+		print('New template position #' + str(position_number))
+
+	except Exception as e:
+		logging.exception('Could not enroll your fingerprint')
+		raise
+
+def display_index_table():
+	"""Displays the index tables of stored fingerprints"""
+	try:
+		logging.info('Currently used templates: %s/%s' % (f.getTemplateCount(), f.getStorageCapacity()))
+		for page in range(0, 4):
+			tableIndex = f.getTemplateIndex(page)
+			s = "";
+			for i in range(0, len(tableIndex)):
+				s = s + "p.%s:index:%s:%s|" % (page, i, tableIndex[i])
+			print(s)
+	except Exception as e:
+		logging.exception('Could not display index table')
+		raise
+
+def init_arg_parse():
+	global parser, usb_device
+	parser = argparse.ArgumentParser(description='Do the given action with fingerprint reader')
+	
+	parser.add_argument("-v", "--verbose", help="Display system parameters of sensor", action="store_true")
+	parser.add_argument("-u", "--usb_device", type=str, help="Sets used usb_device which by default is /dev/ttyUSB0")
+	parser.add_argument("-e", "--enroll", help="Create a new fingerprint model and store it into database", action="store_true")
+	parser.add_argument("-t", "--templates", help="Display current templates count", action="store_true")
+	parser.add_argument("-I", "--index_table", help="Display current index table (templates)", action="store_true")
+	parser.add_argument("-d", "--delete", help="Deletes a template after asking its position to user (first position is 0)", action="store_true")
+	parser.add_argument("-D", "--delete_all_templates", help="Deletes all templates of database", action="store_true")
+	parser.add_argument("-i", "--download_image", help="Downloads image and put it into tempDir", action="store_true")
+	parser.add_argument("-s", "--search", help="Searches for given fingerprint into database and display if found", action="store_true")
```

Comment:
Please add switch `--random` which should use PRNG

Reply:
Dont understand what it should use??? in fact if its about example_generaterandom.py I dont understand the perpose of generating a random number from that peripheral! /dev/random gives it very well!

Reply:
Now... if you want me to add it and it makes sense for you, no problem... maybe I missed something

Reply:
All samples should be contained in one example file. `example_generaterandom.py` is the only sample that is missing.
The ZFM has a hardware accelerated 32-bit PRNG (really cool feature)! It's up to user if he needs it but it should have a sample.